server/schedule/filter/filters.go

```diff
@@ -292,69 +304,105 @@ func (f *StoreStateFilter) Type() string {
 
 // conditionFunc defines condition to determine a store should be selected.
 // It should consider if the filter allows temporary states.
-type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
+type conditionFunc func(*config.PersistOptions, *core.StoreInfo) plan.StatusCode
 
-func (f *StoreStateFilter) isRemoved(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isRemoved(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "tombstone"
-	return store.IsRemoved()
+	if store.IsRemoved() {
+		return plan.StatusStoreUnavailable
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) isDown(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isDown(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "down"
-	return store.DownTime() > opt.GetMaxStoreDownTime()
+	if store.DownTime() > opt.GetMaxStoreDownTime() {
+		return plan.StatusStoreUnavailable
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) isRemoving(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isRemoving(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "offline"
-	return store.IsRemoving()
+	if store.IsRemoving() {
+		return plan.StatusStoreDraining
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) pauseLeaderTransfer(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) pauseLeaderTransfer(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "pause-leader"
-	return !store.AllowLeaderTransfer()
+	if !store.AllowLeaderTransfer() {
+		return plan.StatusStoreBlocked
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) slowStoreEvicted(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) slowStoreEvicted(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "slow-store"
-	return store.EvictedAsSlowStore()
+	if store.EvictedAsSlowStore() {
+		return plan.StatusStoreBlocked
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) isDisconnected(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isDisconnected(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "disconnected"
-	return !f.AllowTemporaryStates && store.IsDisconnected()
+	if !f.AllowTemporaryStates && store.IsDisconnected() {
+		return plan.StatusStoreUnavailable
+	}
+	return plan.StatusOK
 }
 
-func (f *StoreStateFilter) isBusy(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isBusy(opt *config.PersistOptions, store *core.StoreInfo) plan.StatusCode {
 	f.Reason = "busy"
-	return !f.AllowTemporaryStates && store.IsBusy()
+	if !f.AllowTemporaryStates && store.IsBusy() {
+		return plan.StatusStoreUnavailable
```

Comment:
should be `StatusStoreThrottled`?

Reply:
StatusStoreThrottled is aimed at the situation where scheduling is limited by the configuration.